src/core/mac/sub_mac.cpp

```diff
@@ -1055,63 +1039,44 @@ const char *SubMac::StateToString(State aState)
     return kStateStrings[aState];
 }
 
-#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
-const char *SubMac::CslStateToString(CslState aCslState)
-{
-    static const char *const kCslStateStrings[] = {
-        "CslIdle",   // (0) kCslIdle
-        "CslSample", // (1) kCslSample
-        "CslSleep",  // (2) kCslSleep
-    };
-
-    static_assert(kCslIdle == 0, "kCslIdle value is incorrect");
-    static_assert(kCslSample == 1, "kCslSample value is incorrect");
-    static_assert(kCslSleep == 2, "kCslSleep value is incorrect");
-
-    return kCslStateStrings[aCslState];
-}
-#endif
-
 // LCOV_EXCL_STOP
 
 //---------------------------------------------------------------------------------------------------------------------
 // CSL Receiver methods
 
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
-void SubMac::SetCslChannel(uint8_t aChannel)
+bool SubMac::UpdateCsl(uint16_t aPeriod, uint8_t aChannel, otShortAddress aShortAddr, const otExtAddress *aExtAddr)
 {
-    mCslChannel = aChannel;
-}
+    bool diffPeriod  = aPeriod != GetCslPeriod();
+    bool diffChannel = aChannel != GetCslChannel();
+    bool retval      = diffPeriod || diffChannel;
 
-void SubMac::SetCslPeriod(uint16_t aPeriod)
-{
-    VerifyOrExit(mCslPeriod != aPeriod);
+    VerifyOrExit(retval);
+    SetCslChannel(aChannel);
 
-    mCslPeriod = aPeriod;
+    VerifyOrExit(diffPeriod);
+    SetCslPeriod(aPeriod);
+    IgnoreError(Get<Radio>().EnableCsl(aPeriod, aShortAddr, aExtAddr));
 
     mCslTimer.Stop();
-
     if (mCslPeriod > 0)
     {
         mCslSampleTime = TimeMicro(static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance())));
-        mCslState      = kCslSleep;
         HandleCslTimer();
     }
-    else
-    {
-        mCslState = kCslIdle;
 
-        if (mState == kStateCslSample)
-        {
-            IgnoreError(Get<Radio>().Sleep());
-            SetState(kStateSleep);
-        }
-    }
+exit:
+    return retval;
+}
 
-    LogDebg("CSL Period: %u", mCslPeriod);
+void SubMac::SetCslChannel(uint8_t aChannel)
```

Comment:
Now we have `SubMac::UpdateCsl(...)` which is the only way we want these param to be set on `SubMac`. 
So we don't want these to be `public` method on `SubMac` anymore.

How about removing the `SeCslChannel()` and `SetCslPeriod()` method? We can directly set the variables inside `SubMac` itself.

Reply:
Right. Same applies to the getters actually.